tests/unit_tests/test_qvalues.py

```diff
@@ -65,35 +65,44 @@ def asc_scores():
     return scores, target, qvals
 
 
-def test_tdc_descending(desc_scores):
+@pytest.mark.parametrize("dtype", [np.float64, np.uint8, np.int8, np.float32])
+def test_tdc_descending(desc_scores, dtype):
     """Test that q-values are correct for descending scores"""
     scores, target, true_qvals = desc_scores
-    dtypes = [np.float64, np.uint8, np.int8, np.float32]
-    for dtype in dtypes:
-        qvals = tdc(scores.astype(dtype), target, desc=True)
-        np.testing.assert_array_equal(qvals, true_qvals)
+    qvals = tdc(scores.astype(dtype), target, desc=True)
+    np.testing.assert_allclose(qvals, true_qvals, atol=1e-7)
 
-        qvals = tdc(scores, target.astype(dtype), desc=True)
-        np.testing.assert_array_equal(qvals, true_qvals)
+    qvals = tdc(scores, target.astype(dtype), desc=True)
+    np.testing.assert_allclose(qvals, true_qvals, atol=1e-7)
 
 
-def test_tdc_ascending(asc_scores):
+@pytest.mark.parametrize("dtype", [np.float64, np.uint8, np.int8, np.float32])
+def test_tdc_ascending(asc_scores, dtype):
     """Test that q-values are correct for ascending scores"""
     scores, target, true_qvals = asc_scores
-    dtypes = [np.float64, np.uint8, np.int8, np.float32]
-    for dtype in dtypes:
-        qvals = tdc(scores.astype(dtype), target, desc=False)
-        np.testing.assert_array_equal(qvals, true_qvals)
 
-        qvals = tdc(scores, target.astype(dtype), desc=False)
-        np.testing.assert_array_equal(qvals, true_qvals)
+    # Q - June2024 JSP: Since the function is type-checked, what is the purpose
```

Comment:
Question: On this one I am not 100% sure what the test is meant to test. either the function signature is wrong or this test makes no sense :P (if the function only takes arrays of bools, this test should have no point. If it can take more stuff, the signature is wrong)

Reply:
Same below for `test_tdc_non_bool `